store/src/java/com/zimbra/cs/service/account/GetInfo.java

```diff
@@ -477,6 +480,12 @@ private void doDiscoverRights(Element eRights, Account account, Set<Right> right
         DiscoverRights.discoverRights(account, rights, eRights, false);
     }
 
+    private boolean checkIfSpellCheckServiceIsAvailable(Server server) {
+        String[] installedServices = server.getMultiAttr(Provisioning.A_zimbraServiceInstalled);
+        String[] enabledServices = server.getMultiAttr(Provisioning.A_zimbraServiceEnabled);
+        return (Arrays.asList(installedServices).contains("spell") && Arrays.asList(enabledServices).contains("spell"));
+    }
+
```

Comment:
Spell check server does not always run on the mailstore server where an account exists.
For example, there are 2 mailstore server on a system. mailstore1 runs mailboxd and spell. mailstore2 runs mailboxd only and uses spell service which runs on mailstore1.
```
<attr id="267" name="zimbraSpellCheckURL" type="astring" max="256" cardinality="multi" optionalIn="globalConfig,server" flags="serverInherited">
  <desc>URL of the server running the spell checking service.  Multi-valued attribute that allows multiple spell check servers to be specified.  If the request to the first server fails, a request to the second server is sent and so on.</desc>
</attr>

$ zmprov gs `zmhostname` zimbraSpellCheckURL
# name zcs.local
zimbraSpellCheckURL: http://zcs.local:7780/aspell.php
```

Reply:
@rcyarrapothu with this change isn't it still checking for spell service on the local server? I think we need to fetch the servers and check the service in them.

Reply:
If my understanding is correct, `zimbraSpellCheckURL` in the server setting should be checked in the same way as CheckSpellingRequest handling.
```
com.zimbra.cs.service.mail.CheckSpelling
        // Make sure a spell check server is specified
        Provisioning prov = Provisioning.getInstance();
        Server localServer = prov.getLocalServer();
        String[] urls = localServer.getMultiAttr(Provisioning.A_zimbraSpellCheckURL);
        if (ArrayUtil.isEmpty(urls)) {
            log.warn("Unable to check spelling.  No value specified for %s", Provisioning.A_zimbraSpellCheckURL);
            return unavailable(response);
        }
```
`zimbraServiceEnabled spell` just shows whether Spell Service runs on the server or not. If `zimbraServiceEnabled spell` is set on a server, httpdis up on 7780 port at `zmcontrol start`. It does not mean that whether or not "Spell check function is available for the account who belongs to the server".

Reply:
@k-kato that's correct. Have modified the logic to consider "zimbraSpellCheckURL" instead of zimbraServiceEnabled/zimbraServiceInstalled. So now, setting the boolean value to false whenever zimbraSpellCheckURL returns empty on  the primary server. Thanks for your suggestion!

Reply:
@dasiyogesh that's was the commit while we were still discussing the approach. The latest commit checks for the zimbraSpellCheckURL.